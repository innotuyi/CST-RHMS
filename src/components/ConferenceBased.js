import React, { useState } from "react";
import axios from 'axios';
import { useHistory } from "react-router-dom";
import {
  Row,
  Col,
  Container,
  Form,
  Button,
  Jumbotron,
} from "react-bootstrap";
import Header from "./Header";

const ConferenceBased = () => {
  const history = useHistory();
  const [name, setName] = useState("");
  const [author, setAuthor] = useState("");
  const [theme, setTheme] = useState("");
  const [organizer, setOrganizer] = useState("");
  const [confplace, setConfPlace] = useState("");
  const [editor, setEditor] = useState("");
  const [chapterPage, setChapterPage] = useState("");
  const [isbn, setISBN] = useState("");

  const submitHandler = (e) => {
    e.preventDefault();
    const payload = {
      'confer_name': name, 
       'author':author,
       'theme':theme,
       'organizer':organizer,
       'place':confplace,
       'editor':editor,   
       'no_of_pages':chapterPage,
       'ISBN':isbn,       
       'research':'PRJA1'}
axios.post('http://127.0.0.1:8000/conference/', payload)
.then(function (response) {
console.log('Successful response: ',  response);
})
.catch(function (error) {
console.log('Error response: ', error);
});

history.push('/Conference');

    console.warn(
      name,
      theme,
      organizer,
      editor,
      chapterPage,
      confplace,
      isbn
    );
  };

  

    return (
      <div
        style={{
          background: "linear-gradient(to bottom, white, 84%, #1089AF)",
        }}
      >
        <Header />
        <Container>
          <Row className="d-flex justify-content-center arighn-item-center">
            <Col className="d-flex justify-content-center">
              <Jumbotron
                style={{
                  background:
                    "linear-gradient(to bottom, white , 80%, #1089AF )",
                  borderRadius: "10px",
                  width: "90%",
                }}
              >
                <Form className="form" style={{}} onSubmit={submitHandler}>
                  <h2 className="text-center mb-5 text-secondary">
                    Conference Based Research Information
                  </h2>
                  <Form.Group>
                    <Form.Label>Conference name</Form.Label>
                    <Form.Control
                      required
                      type="text"
                      id="name"
                      value={name}
                      onChange={(e) => setName(e.target.value)}
                      placeholder="Enter The Name Of The Conference"
                    />
                  </Form.Group>
                      <Form.Group>
                    <Form.Label> Author</Form.Label>
                    <Form.Control
                      required
                      type="email"
                      id="name"
                      value={author}
                      onChange={(e) => setAuthor(e.target.value)}
                      placeholder="Enter Author Name"
                    />
                  </Form.Group>
                  <Form.Group>
                    <Form.Label>Conference Theme</Form.Label>
                    <Form.Control
                      required
                      type="text"
                      id="theme"
                      value={theme}
                      onChange={(e) => setTheme(e.target.value)}
                      placeholder="Enter Conference Theme "
                    />
                  </Form.Group>
                  <Form.Group>
                    <Form.Label>Orgnizer's Name</Form.Label>
                    <Form.Control
                      required
                      type="text"
                      id="organizer"
                      value={organizer}
                      onChange={(e) => setOrganizer(e.target.value)}
                      placeholder="Enter Orgnizer's Name"
                    />
                  </Form.Group>
                  <Row>
                    <Col>
                      {/* <Form.Group controlId="dob">
                        <Form.Label>Select Conference Date</Form.Label>
                        <Form.Control
                          type="date"
                          name="dos"
                          required
                          id="date"
                          value={confDate}
                          onChange={(e) => setConfDate(e.target.value)}
                          placeholder="Conference Date"
                        />
                      </Form.Group> */}
                                 <Form.Group>
                        <Form.Label> Place</Form.Label>
                        <Form.Control
                          type="text"
                          name="dos"
                          required
                          id="place"
                          value={editor}
                          onChange={(e) => setEditor(e.target.value)}
                          placeholder="Enter Conference Location"
                        />
                      </Form.Group>
                    </Col>
                    <Col>
                    <Form.Group>
                    <Form.Label>Editor</Form.Label>
                    <Form.Control
                      required
                      type="text"
                      id="name"
                      value={confplace}
                      onChange={(e) => setConfPlace(e.target.value)}
                      placeholder="Enter The Name Of The Conference"
                    />
                  </Form.Group>
           
                    </Col>
                  </Row>
                  <Row>
                    <Col>
                      <Form.Group controlId="">
                        <Form.Label>Chapter Pages</Form.Label>
                        <Form.Control
                          type="Number"
                          name=""
                          required
                          id="page"
                          value={chapterPage}
                          onChange={(e) => setChapterPage(e.target.value)}
                          placeholder="Enter The Number Of Pages"
                        />
                      </Form.Group>
                    </Col>
                    <Col>
                      <Form.Group>
                        <Form.Label>ISBN</Form.Label>
                        <Form.Control
                          required
                          type="Number"
                          name=""
                          id="isbn"
                          value={isbn}
                          onChange={(e) => setISBN(e.target.value)}
                          placeholder="Enter ISBN"
                        />
                      </Form.Group>
                    </Col>
                  </Row>
                  <Col className="text-center">
                    <Button className="" type="submit">
                      Create
                    </Button>
                  </Col>
                </Form>
              </Jumbotron>
            </Col>
          </Row>
        </Container>
      </div>
    );
}

export default ConferenceBased
