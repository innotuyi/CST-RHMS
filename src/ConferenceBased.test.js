import React from "react";
import ConferenceBased from "./components/ConferenceBased";

import { configure, shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

let wrapper;
const name = "Rwigamba";

beforeEach(() => {
  wrapper = shallow(<ConferenceBased />);
});

describe("Conference Based Tests", () => {

  it("matches Snapshot Collectly", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

it("should have All Inputs", () => {
  expect(wrapper.find("#name")).toHaveLength(1);
  expect(wrapper.find("#theme")).toHaveLength(1);
  expect(wrapper.find("#organizer")).toHaveLength(1);
  expect(wrapper.find("#date")).toHaveLength(1);
  expect(wrapper.find("#place")).toHaveLength(1);
  expect(wrapper.find("#page")).toHaveLength(1);
  expect(wrapper.find("#isbn")).toHaveLength(1);
});

it("should have a Button  Component", () => {
  expect(wrapper.find("Button")).toHaveLength(1);
  expect(wrapper.find("Button").text()).toEqual("Create");
});

it("Should Conference Name correctly on Change", () => {
  wrapper.find("#name").simulate("change", {
    target: {
      value: name,
    },
  });
  expect(wrapper.find("#name").props().value).toEqual(name);
});

it("Should capture Conference Theme correctly on Change", () => {
  wrapper.find("#theme").simulate("change", {
    target: {
      value: name,
    },
  });
  expect(wrapper.find("#theme").props().value).toEqual(name);
});

it("Should capture Orgnizer's Name correctly on Change", () => {
  wrapper.find("#organizer").simulate("change", {
    target: {
      value: name,
    },
  });
  expect(wrapper.find("#organizer").props().value).toEqual(name);
});

it("Should capture Conference Location correctly on Change", () => {
  wrapper.find("#place").simulate("change", {
    target: {
      value: name,
    },
  });
  expect(wrapper.find("#place").props().value).toEqual(name);
});

it("Should capture Conference Date correctly on Change", () => {
  wrapper.find("#date").simulate("change", {
    target: {
      value: "2021-06-16",
    },
  });
  expect(wrapper.find("#date").props().value).toEqual("2021-06-16");
});

it("Should capture Chapter Pages correctly on Change", () => {
  wrapper.find("#page").simulate("change", {
    target: {
      value: "20",
    },
  });
  expect(wrapper.find("#page").props().value).toEqual("20");
});

it("Should capture ISBN correctly on Change", () => {
  wrapper.find("#isbn").simulate("change", {
    target: {
      value: "20",
    },
  });
  expect(wrapper.find("#isbn").props().value).toEqual("20");
});

});
