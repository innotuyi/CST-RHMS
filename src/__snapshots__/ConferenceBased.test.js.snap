// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conference Based Tests matches Snapshot Collectly 1`] = `
<div
  style={
    Object {
      "background": "linear-gradient(to bottom, white, 84%, #1089AF)",
    }
  }
>
  <Header />
  <Container
    fluid={false}
  >
    <Row
      className="d-flex justify-content-center arighn-item-center"
      noGutters={false}
    >
      <Col
        className="d-flex justify-content-center"
      >
        <Jumbotron
          fluid={false}
          style={
            Object {
              "background": "linear-gradient(to bottom, white , 80%, #1089AF )",
              "borderRadius": "10px",
              "width": "90%",
            }
          }
        >
          <Form
            className="form"
            inline={false}
            onSubmit={[Function]}
            style={Object {}}
          >
            <h2
              className="text-center mb-5 text-secondary"
            >
              Conference Based Research Information
            </h2>
            <FormGroup>
              <FormLabel
                column={false}
                srOnly={false}
              >
                Conference name
              </FormLabel>
              <FormControl
                id="name"
                onChange={[Function]}
                placeholder="Enter The Name Of The Conference"
                required={true}
                type="text"
                value=""
              />
            </FormGroup>
            <FormGroup>
              <FormLabel
                column={false}
                srOnly={false}
              >
                Conference Theme
              </FormLabel>
              <FormControl
                id="theme"
                onChange={[Function]}
                placeholder="Enter Conference Theme "
                required={true}
                type="text"
                value=""
              />
            </FormGroup>
            <FormGroup>
              <FormLabel
                column={false}
                srOnly={false}
              >
                Orgnizer's Name
              </FormLabel>
              <FormControl
                id="organizer"
                onChange={[Function]}
                placeholder="Enter Orgnizer's Name"
                required={true}
                type="text"
                value=""
              />
            </FormGroup>
            <Row
              noGutters={false}
            >
              <Col>
                <FormGroup
                  controlId="dob"
                >
                  <FormLabel
                    column={false}
                    srOnly={false}
                  >
                    Select Conference Date
                  </FormLabel>
                  <FormControl
                    id="date"
                    name="dos"
                    onChange={[Function]}
                    placeholder="Conference Date"
                    required={true}
                    type="date"
                    value=""
                  />
                </FormGroup>
              </Col>
              <Col>
                <FormGroup>
                  <FormLabel
                    column={false}
                    srOnly={false}
                  >
                     Place
                  </FormLabel>
                  <FormControl
                    id="place"
                    name="dos"
                    onChange={[Function]}
                    placeholder="Enter Conference Location"
                    required={true}
                    type="text"
                    value=""
                  />
                </FormGroup>
              </Col>
            </Row>
            <Row
              noGutters={false}
            >
              <Col>
                <FormGroup
                  controlId=""
                >
                  <FormLabel
                    column={false}
                    srOnly={false}
                  >
                    Chapter Pages
                  </FormLabel>
                  <FormControl
                    id="page"
                    name=""
                    onChange={[Function]}
                    placeholder="Enter The Number Of Pages"
                    required={true}
                    type="Number"
                    value=""
                  />
                </FormGroup>
              </Col>
              <Col>
                <FormGroup>
                  <FormLabel
                    column={false}
                    srOnly={false}
                  >
                    ISBN
                  </FormLabel>
                  <FormControl
                    id="isbn"
                    name=""
                    onChange={[Function]}
                    placeholder="Enter ISBN"
                    required={true}
                    type="Number"
                    value=""
                  />
                </FormGroup>
              </Col>
            </Row>
            <Col
              className="text-center"
            >
              <Button
                active={false}
                className=""
                disabled={false}
                type="submit"
                variant="primary"
              >
                Create
              </Button>
            </Col>
          </Form>
        </Jumbotron>
      </Col>
    </Row>
  </Container>
</div>
`;
